arangod/Aql/ExecutionBlock.h

```diff
@@ -83,6 +83,11 @@ class ExecutionBlock {
   /// @brief batch size value
   static constexpr inline size_t DefaultBatchSize() { return 1000; }
 
+  /// @brief Number to use when we skip all. Should really be inf, but don't
+  /// use something near std::numeric_limits<size_t>::max() to avoid overflows
+  /// in calculations.
+  static constexpr inline size_t SkipAllSize() { return 1000000000; }
```

Comment:
It is a value that is used for "infinity" when everything should be skipped. Previously, we would skip at most 1000 documents at a time. It is not a limit that has to be (or is) obeyed, just the amount we try to skip at a time when we want to skip everything. In case there is more to skip than the constant, it will just be called again until everything is done, so nothing bad happens.

In other words, everything should work fine if we set this to at least 1, only performance suffers if its too small, and we might overflow when we come near `std::numeric_limits<size_t>::max()`.

Reply:
I'll extend the comment a little to make that clear.